hooks/useIsMobile.ts

```diff
@@ -0,0 +1,20 @@
+import { useEffect, useState } from 'react'
+
+export const useIsMobile = () => {
+    const [isMobile, setIsMobile] = useState<boolean>(false)
+
+    useEffect(() => {
+        const handleIsMobile = () => {
+            if (window.innerWidth <= 1024) {
+                setIsMobile(true)
+            } else {
+                setIsMobile(false)
+            }
+        }
+        window.addEventListener('resize', handleIsMobile)
```

Comment:
I guess this is used mostly in development mode so we should make a condition to avoid in production mode.